src/net_processing.cpp

```diff
@@ -3987,14 +3998,14 @@ bool PeerLogicValidation::SendMessages(CNode* pto)
                     // up processing to happen after the download times out
                     // (with a slight delay for inbound peers, to prefer
                     // requests to outbound peers).
-                    RequestTx(&state, txid, nNow);
```

Comment:
`RequestTx()` does the same as the two lines below, but performs `.size()` checks on the tx maps and ensures the txid is in `m_tx_announced`. Are we not just calling that because we don't care to do those checks as belt-and-suspenders, or do we specifically want to avoid populating `m_tx_announced`? I guess those checks are safe to avoid given txid was already in `tx_process_time`, but seems like it wouldn't hurt to do them anyway.

Reply:
`RequestTx()` is inappropriate here and was a bug -- see @ajtowns' comment at https://github.com/bitcoin/bitcoin/pull/15834#discussion_r276112938

Reply:
Oops, sorry I missed that. Thanks for the pointer.